kratos/tests/containers/test_data_value_container.cpp

```diff
@@ -38,5 +42,50 @@ KRATOS_TEST_CASE_IN_SUITE(DataValueContainer, KratosCoreFastSuite) {
   for (std::size_t i = 0; i < distances.size(); i++)
     KRATOS_CHECK_EQUAL(distances[i], original_distances[i]);
 }
+
+/**
+ * This test checks recursively the proper functioning of the DataValueContainer
+ */
+KRATOS_TEST_CASE_IN_SUITE(DataValueContainerRecursive, KratosCoreFastSuite) {
+  DataValueContainer container;
+  Vector original_distances(4);
+  original_distances[0] = 0.00;
+  original_distances[1] = 0.10;
+  original_distances[2] = 0.20;
+  original_distances[3] = 0.30;
+
+  const std::size_t number_recursions = 1000;
```

Comment:
Why you should repeat this 1000 times?

Reply:
The number is just arbitrary large, I was trying to check that the memory is not corrupted doing the same several times

Reply:
You will see a memory collapse by valgrind in few iterations (say 1 to 3) and usually such a stress tests are for finding concurrency problems.

Reply:
OK, reverting